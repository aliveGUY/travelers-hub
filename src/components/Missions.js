import React from 'react';
import { useSelector } from 'react-redux/es/exports';

export default function Missions() {
  const missions = useSelector((state) => state.missions);
  return (
    <div className="Missions-wrapper">
      <table className="Missions-table">
        <tbody>
          <tr>
            <th>Mission</th>
            <th className="Missions-description">Description</th>
            <th>Status</th>
            <th>Interaction</th>
          </tr>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <th>{mission.mission_name}</th>
              <th className="Missions-description">{mission.description}</th>
              <th>Status</th>
              <th>
                <button type="button">submit</button>
              </th>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}
