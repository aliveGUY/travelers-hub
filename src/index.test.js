import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import Missions from './components/Missions';

const mockStore = configureStore([]);

describe('Missions testing', () => {
  const mission = [
    {
      mission_id: 0,
      mission_name: 'Mission Name',
      description: 'Mission Description',
    },
  ];
  test('REnder Missions', () => {
    const store = mockStore({
      missions: mission,
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Should dispatch MissionReserve action', () => {
    const store = mockStore({
      missions: mission,
    });
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Join Mission'));
    const firedAction = store.getActions();
    expect(...firedAction).toEqual({
      payload: 0,
      type: 'Missions/joinMission',
    });
  });
});
