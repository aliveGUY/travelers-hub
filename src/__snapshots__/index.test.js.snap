// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions testing REnder Missions 1`] = `
<div
  className="Missions-wrapper"
>
  <table
    className="Missions-table"
  >
    <tbody>
      <tr>
        <th>
          Mission
        </th>
        <th
          className="Missions-description"
        >
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Interaction
        </th>
      </tr>
      <tr>
        <th>
          Mission Name
        </th>
        <th
          className="Missions-description"
        >
          Mission Description
        </th>
        <th>
          <p
            className="Mission-incative"
          >
            NOT A MEMBER
          </p>
        </th>
        <th>
          <button
            className="Mission-button "
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
